utensor_cgen/_pbparser_impl.py

```diff
@@ -89,26 +90,30 @@ def _parse_graph_nodes(graph_def):
   - thought the output tensor names is irrelevent for TensorFlow, but it
     is neccessary for uTensor
   """
+  OperationInfo = namedtuple('OperationInfo',
+                             field_names=['input_tensor', 'output_tensor',
+                                          'op_type', 'output_content'])
   graph = Graph()
   with graph.as_default():  # pylint: disable=E1129
     import_graph_def(graph_def, name="")
-  graph_info = defaultdict(lambda: {"output_content": {}})
+  graph_info = {}
   with Session(graph=graph):
     for node in graph_def.node:
       op = graph.get_operation_by_name(node.name)
-      op_info = graph_info[node.name]
-      op_info["input_tensor"] = [(t.name, t.dtype, _parse_shape(t.shape)) for t in op.inputs]
-      op_info["output_tensor"] = [(t.name, t.dtype, _parse_shape(t.shape)) for t in op.outputs]
-      op_info["op_type"] = node.op
+      input_tensor = [(t.name, t.dtype, _parse_shape(t.shape)) for t in op.inputs]
+      output_tensor = [(t.name, t.dtype, _parse_shape(t.shape)) for t in op.outputs]
+      op_type = node.op
+      output_content = {}
       if node.op in ["Const"]:
-        for out_tensor, _, _ in op_info["output_tensor"]:
-          tensor = graph.get_tensor_by_name(out_tensor)
-          op_info["output_content"][tensor.name] = tensor.eval()
+        for tensor_name, _, _ in output_tensor:
+          tensor = graph.get_tensor_by_name(tensor_name)
+          output_content[tensor_name] = tensor.eval()
+      graph_info[node.name] = OperationInfo(input_tensor, output_tensor, op_type, output_content)
   return graph_info
 
 
 def _parse_graph_layers(graph_def):
```

Comment:
Is this necessary? Can't we just trust the output names from the CLI input

Reply:
See my comment below.